app/assets/javascripts/discourse/app/components/admin-post-menu.gjs

```diff
@@ -0,0 +1,231 @@
+import Component from "@glimmer/component";
+import { inject as service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import { fn } from "@ember/helper";
+import and from "truth-helpers/helpers/and";
+import or from "truth-helpers/helpers/or";
+import not from "truth-helpers/helpers/not";
+import { action } from "@ember/object";
+import concatClass from "discourse/helpers/concat-class";
+
+export default class AdminPostMenu extends Component {
+  @service currentUser;
+  @service siteSettings;
+  @service store;
+  @service adminPostMenuButtons;
+
+  <template>
+    <ul>
+      {{#if this.currentUser.staff}}
+        <li>
+          <DButton
+            @label="review.moderation_history"
+            @icon="list"
+            class="btn btn-transparent moderation-history"
+            @href={{this.reviewUrl}}
+          />
+        </li>
+      {{/if}}
+
+      {{#if (and this.currentUser.staff (not @data.transformedPost.isWhisper))}}
+        <li>
+          <DButton
+            @label={{if
+              @data.transformedPost.isModeratorAction
+              "post.controls.revert_to_regular"
+              "post.controls.convert_to_moderator"
+            }}
+            @icon="shield-alt"
+            class={{concatClass
+              "btn btn-transparent toggle-post-type"
+              (if @data.transformedPost.isModeratorAction "btn-success")
+            }}
+            @action={{fn this.topicAction "togglePostType"}}
+          />
+        </li>
+      {{/if}}
```

Comment:
yes same it was the widget code, I know it can be simplified, but don't want to change too much, I just wanted to have use cases for each cases we have in the codebase to be sure to have a decent solution in each case